comp/netflow/testutil/freeport.go

```diff
@@ -9,36 +9,20 @@ package testutil
 
 import (
 	"net"
-	"strconv"
 )
 
 // GetFreePort finds a free port to use for testing.
-func GetFreePort() uint16 {
-	var port uint16
-	for i := 0; i < 5; i++ {
-		conn, err := net.ListenPacket("udp", ":0")
-		if err != nil {
-			continue
-		}
-		conn.Close()
-		port, err = parsePort(conn.LocalAddr().String())
-		if err != nil {
-			continue
-		}
-		return port
-	}
-	panic("unable to find free port for starting the trap listener")
-}
-
-func parsePort(addr string) (uint16, error) {
-	_, portString, err := net.SplitHostPort(addr)
+// Borrowed from: https://github.com/phayes/freeport/blame/master/freeport.go#L8-L20
+func GetFreePort() (uint16, error) {
```

Comment:
Note: this new version of the freeport is more reliable. The previously one was sometimes leading to "port already used"

Reply:
It's a pity we can't pass the port to goflow2, since even this could be flaky if something else claims whatever port we find between the call to GetFreePort and the actual start of the goflow2 instance.

I guess really the ideal would be if goflow2 provided a way to mock the port so that we didn't need to connect to anything at all and could just send data directly to the goflow2 listener...

Reply:
Also we have a duplicate of this function in the snmp traps tests, and I'm not sure if there are others elsewhere - maybe this should move to a separate utility somewhere?